src/passes/OptimizeCasts.cpp

```diff
@@ -453,20 +437,32 @@ struct BestCastFinder : public LinearExecutionWalker<BestCastFinder> {
   void visitRefCast(RefCast* curr) { handleRefinement(curr); }
 
   void handleRefinement(Expression* curr) {
+    auto* teeFallthrough = Properties::getFallthrough(
+      curr, options, *getModule(), Properties::FallthroughBehavior::NoTeeBrIf);
+    if (auto* set = teeFallthrough->dynCast<LocalSet>()) {
+      if (set->isTee()) {
+        updateBestCast(curr, set->index);
+      }
+    }
     auto* fallthrough = Properties::getFallthrough(curr, options, *getModule());
```

Comment:
```suggestion
    auto* fallthrough = Properties::getFallthrough(teeFallthrough, options, *getModule());
```
This saves a little work: we looked a little the first time, and the second time we don't need to re-do that work, and just continue from there.